### PR TITLE
Fix handling of file edit failures

### PR DESCRIPTION
**Pull request by nwalters512**
I noticed on https://github.com/PrairieLearn/PrairieLearn/pull/7966#discussion_r1230099379 that the previous way of checking for failures (inspecting individual job names/statuses) was broken by #7949. This PR fixes that by using the new server job `data` store to persist structured state that we can then read back later.

### CODE REVIEW

File: apps/prairielearn/src/lib/editors.js [hunk 1]
```diff
@@ -167,10 +167,18 @@ class Editor {
               }
 
               try {
+                // Remember that we attempted a push. If the push fails, we'll retain
+                // the `false` value.
+                job.data.pushed = false;
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/editors.js:172]: See https://github.com/PrairieLearn/PrairieLearn/pull/7966#discussion_r1244479860 for a discussion of if we should use separate flags for "did attempt" and "did succeed".

Comment (mwest1066) [reply apps/prairielearn/src/lib/editors.js:172]: As I commented there, I (weakly) prefer separate flags.

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/src/lib/editors.js (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/lib/editors.js]: I'll take it! Done in #343b850.

File: apps/prairielearn/src/lib/editors.js [hunk 2]
```diff
@@ -181,7 +187,13 @@ class Editor {
                 await cleanAndResetRepository(this.course, gitEnv, job);
               }
 
+              job.data.syncAttempted = false;
```
Comment (mwest1066) [inline apps/prairielearn/src/lib/editors.js:190]: Should this be `true`?

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/src/lib/editors.js (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/lib/editors.js:190]: Yes it should!

```suggestion
              job.data.syncAttempted = true;
```

File: apps/prairielearn/src/lib/editors.js [hunk 3]
```diff
@@ -184,7 +190,13 @@ class Editor {
                 await cleanAndResetRepository(this.course, gitEnv, job);
               }
 
+              job.data.syncAttempted = true;
+
               await syncCourseFromDisk(this.course, startGitHash, job);
+
+              // As with `job.data.pushAttempted` above, we'll check this flag
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/editors.js:197]:
```suggestion
              // As with `job.data.pushSucceeded` above, we'll check this flag
```

**Review by mwest1066 — APPROVED** [review]

**Pull request merged**